src/window.rs

```diff
@@ -1511,6 +1511,49 @@ unsafe impl HasRawDisplayHandle for Window {
     }
 }
 
+/// An error that may be generated when requesting window state.
+// TODO(madsmtm): Split this into further variants.
+#[derive(Debug)]
+pub enum WindowError {
+    /// The operation is not supported by the backend.
+    NotSupported(NotSupportedError),
+    /// The OS cannot perform the operation.
+    Os(OsError),
+}
+
+impl fmt::Display for WindowError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        match self {
+            Self::NotSupported(e) => write!(f, "window operation failed: {e}"),
```

Comment:
Could just name it a `RequestError`. I'm also not a fan of `TODO` items in top level stuff, could just open an issue.